4.4.js

```diff
@@ -0,0 +1,29 @@
+function deepEqual(a, b) {
+    if (a === b) {
+        return true;
+    }
+
+    if (typeof a != "object" || a == null || typeof b == "object" || b != null) {
+        return false;
+    }
+
+    var propsA = 0,

```

Comment:
prosze nie korzystaj z single var pattern (to jedyny pattern z javascript patterns, ktorego stosowania nie zalecam). dlaczego?

w es6 wprowadzonych zostalo pare slow kluczowych oznaczajacych kontenery na dane (napisalbym zmienne, ale to nie tylko zmienne) - var, let, const.

var na poczatku od razu pokazuje Ci, ze cos jest zmienna. jesli gdzies przez przypadek zamienisz przecinek na srednik otrzymasz zmienna globalna.